### PR TITLE
Fix/62 Tab Title editable area fix

### LINKED ISSUE #62: Tab Title is editable from below the tab!
**Describe the bug**
It is noted in #29 that the tabs titles are now no more editable from their own location, however, it is now displaying below the tab! See https://github.com/10up/publisher-media-kit/pull/29#issuecomment-946592607

editable-tab-title-changed-location

It should be editable from their own location as per the upstream tabs.

10up-tabs-block-title
[image link](https://www.screencast.com/t/xUpH6Iqn9Vt)

**Steps to Reproduce**
1. Go to PMK Media Kit page.
2. Scroll to "Our Rates" section.
3. See the tab titles are editable from below the tab, unlike upstream.

**Expected behavior**
The tabs titles from the PMK plugin should be editable like upstream tabs.

### PR DESCRIPTION
**Pull request by faisal-alvi**
### Description of the Change

The tab title is now fixed to be editable from their own location as per the upstream tabs. To do so, the header of the tabs section (for example, "Our Rates") is moved on a separate line in the (backend area only) instead of displaying it inline with the tabs.

### Verification Process

- Go to the Media Kit page in the back end.
- Scroll to the "Our Rates" section.
- See the tab titles are editable from its own location like upstream.

### Checklist:

- [ ] I have read the [**CONTRIBUTING**](/CONTRIBUTING.md) document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests passed.

### Applicable Issues

Closes #62.

### Changelog Entry

> Fix - Tab Title is editable from below the tab

### CODE REVIEW

File: assets/css/frontend/components/tabs.css [hunk 1]
```diff
@@ -1,3 +1,32 @@
+body:not(.wp-admin) div.wp-block-group.pmk-tabs-table {
+
+	& .tabs-header {
```
Comment (peterwilsoncc) [inline assets/css/frontend/components/tabs.css:3]:
```suggestion
	.tabs-header {
```
Nit: `&` isn't required for descendant selectors (can't see a requirement in style guide but correct me if wrong)

:1234: Applies in multiple locations.

File: assets/css/frontend/components/tabs.css [hunk 2]
```diff
@@ -12,35 +41,23 @@ div.wp-block-group.pmk-tabs-table {
 	}
 
 	& .tab-title {
+		border-bottom: 1px solid var(--pmk-global-color-border);
 		font-size: var(--pmk-heading-2-font-size);
 		font-weight: var(--pmk-font-weight-normal);
 		line-height: var(--pmk-heading-2-line-height);
-		margin: 0 calc(var(--pmk-spacing-internal-medium) * 2) 0 0;
+		margin: 0 calc(var(--pmk-spacing-internal-medium) * 2) 15px 0;
```
Comment (peterwilsoncc) [inline assets/css/frontend/components/tabs.css]: `15px` is a magic number, is there are CSS variable or relative unit that can be used in its place?

File: assets/css/frontend/components/tabs.css [hunk 3]
```diff
@@ -1,3 +1,32 @@
...
+
+	& .tab-title {
+		border: 0;
+		margin-bottom: 0;
+
+		&::after {
+			background-color: var(--pmk-global-color-border);
+			content: "";
+			display: block;
+			height: 20px;
```
Comment (peterwilsoncc) [inline assets/css/frontend/components/tabs.css]: `20px` is a magic number, is there are CSS variable or relative unit that can be used in its place?

**Review by peterwilsoncc — COMMENTED** [review]
Generally looks good, I've asked a few questions inline but feel free to push back as I know you have more experience on this project than I.

**Manual testing notes**

In the edit view, the text of the active tab is substantially smaller than the text of inactive tabs. On the front end view, the font is always the same size -- that of the inactive tabs in the edit view.

tabs-small-active-heading

**Comment by faisal-alvi** [comment]
Thanks for the review @peterwilsoncc. @jeffpaul can we have any FEE available to work on these suggestions?

File: assets/css/frontend/components/tabs.css (hunk 1 above)
Comment (kdo) [reply assets/css/frontend/components/tabs.css:3]: @peterwilsoncc It's required as per stylelint `selector-nested-pattern` rule.
By the way, nice to meet you! 😅

File: assets/css/frontend/components/tabs.css (hunk 2 above)
Comment (kdo) [reply assets/css/frontend/components/tabs.css]: @peterwilsoncc looks like it's coming from: https://github.com/10up/publisher-media-kit/pull/63/files#diff-c21443a0d8d49c507edc76dc2011f74419336d04f418bae4393c1e8d2e39d9e8L25
I'm going to make this 50%, this is the decorative border on the right of the titles. Thanks!

Comment (kdo) [reply assets/css/frontend/components/tabs.css]: @peterwilsoncc woops, previous comment is on the wrong line. Please ignore.

File: assets/css/frontend/components/tabs.css (hunk 3 above)
Comment (kdo) [reply assets/css/frontend/components/tabs.css]: @peterwilsoncc looks like it's coming from: https://github.com/10up/publisher-media-kit/pull/63/files#diff-c21443a0d8d49c507edc76dc2011f74419336d04f418bae4393c1e8d2e39d9e8L25
I'm going to make this 50%, this is the decorative border on the right of the titles. Thanks!

**Comment by kdo** [comment]
> Generally looks good, I've asked a few questions inline but feel free to push back as I know you have more experience on this project than I.
> 
> **Manual testing notes**
> 
> In the edit view, the text of the active tab is substantially smaller than the text of inactive tabs. On the front end view, the font is always the same size -- that of the inactive tabs in the edit view.
> 
> tabs-small-active-heading

@peterwilsoncc Looking at this issue, this is from the 10up tabs library, the font-size for the active tab is set to 13px. Should the plugin editor.css for the tabs be in sync with the 10up tabs release? I didn't edit because I'm not familiar with the project.

I've pushed some tweaks for the other CR comments.

Thanks!

**Comment by faisal-alvi** [comment]
@kdo thanks for the help! I will go ahead and work on the fix in the upstream. Till then, we can hold on to this PR.

**Review by peterwilsoncc — APPROVED** [review]
Code looks good to me, ready for QA.

**Pull request merged**